### DISCUSSION TITLE
gcode asterisks

### DISCUSSION THREAD

**Discussion by tocguy** [supermerill/SuperSlicer#3552] [Q&A]
While printing via Octoprint, in the terminal window I'm seeing peculiar gcode formatting. By peculiar I mean most lines are appended with an asterisk followed by two to three digits. Why is that?
```
Send: N826 G1 X152.72 Y92.342 E6.34115*103
Recv: ok
Send: N827 G1 X152.588 Y92.342 E6.3493*105
Recv: ok
Send: N828 G1 X147.72 Y88.933 E6.71668*102
Recv: ok
Send: N829 G1 X147.72 Y96.427 E7.17992*107
Recv: ok```
Plz advise...

TIA

**Comment by ghost** [comment]
OctoPrint includes the checksum. It's a bit-wise xor of every character.
CRC - Cyclic Redundancy Check
